src/pages/index.tsx

```diff
@@ -83,30 +56,17 @@ const Index: FC = () => {
     const onSelect = (tab: Tab) => {
         setActiveTab(tab);
     };
-    useEffect(() => {
-        if ('tasks' in response) {
-            const tasks = updateTasksStatus(response.tasks);
-            tasks.sort((a: task, b: task) => +a.endsOn - +b.endsOn);
-            tasks.sort(
-                (a: task, b: task) =>
-                    STATUS_ORDER.indexOf(a.status) -
-                    STATUS_ORDER.indexOf(b.status)
-            );
-            const taskMap: any = [];
-            tasks.forEach((item) => {
-                if (item.status in taskMap) {
-                    taskMap[item.status] = [...taskMap[item.status], item];
-                } else {
-                    taskMap[item.status] = [item];
-                }
-            });
-            setFilteredTask(taskMap);
-        }
 
-        return () => {
-            setFilteredTask([]);
-        };
-    }, [isLoading, response]);
+    const tasksGroupedByStatus = updateTasksStatus(
+        response?.tasks || []
+    ).reduce((acc: Record<string, task[]>, curr: task) => {
+        return acc[curr.status as keyof task]
+            ? {
+                  ...acc,
+                  [curr.status]: [...acc[curr.status as keyof task], curr],
+              }
+            : { ...acc, [curr.status]: [curr] };
+    }, {});
 
```

Comment:
Thanks for the refactor

Reply:
Will address that in further iterations, Bharti has also wrote some util function for this, I will check it with that in the next PR

Reply:
Got it.